src/APIBasedAuth.ts

```diff
@@ -1,57 +1,58 @@
-import axios, { AxiosError, AxiosInstance, AxiosResponse } from 'axios';
+import axios, { AxiosInstance, AxiosResponse } from 'axios';
+import { window } from './globals';
+import { DEFAULT_BASE_URL, formatAxiosError } from './utils/helper';
 
-// exporting just for jest test coverage
-export const DEFAULT_BASE_URL = 'https://apps.us.lifeomic.com/auth/v1/api';
-
-// exporting just for jest test coverage
-export const formatAxiosError = <ErrorResponse = any>(
-  error: AxiosError<ErrorResponse> | unknown
-) => {
-  if (!axios.isAxiosError(error)) {
-    throw new AxiosError(String(error));
-  }
-  throw error.response?.data;
-};
+export const API_AUTH_STORAGE_KEY = 'lo-app-tools-api-auth';
+const SESSION_KEYS = ['session', 'username'];
+const TOKEN_KEYS = ['accessToken', 'idToken', 'refreshToken'];
 
 class APIBasedAuth {
   private client: AxiosInstance;
-  private clientId: APIBasedAuth.Config['clientId'];
+  private clientOptions: APIBasedAuth.Config;
   private session?: APIBasedAuth.Session;
 
-  constructor({ clientId, ...options }: APIBasedAuth.Config) {
+  constructor({
+    baseURL,
+    clientId,
+    storage,
+    storageKeys = {}
+  }: APIBasedAuth.Config) {
     if (!clientId) {
       throw new Error('APIBasedAuth param clientId is required');
     }
 
-    this.client = axios.create({
-      ...options,
-      baseURL: options.baseURL || DEFAULT_BASE_URL
-    });
+    this.client = axios.create({ baseURL: baseURL || DEFAULT_BASE_URL });
 
-    this.clientId = clientId;
+    this.clientOptions = {
+      clientId,
+      storage: storage || window.localStorage,
+      storageKeys
+    };
   }
 
-  public confirmPasswordlessAuth(
+  public async confirmPasswordlessAuth(
     baseInput: Omit<APIBasedAuth.VerifyPasswordlessAuthData, 'clientId'>
   ) {
     const input = {
       ...baseInput,
       // if session or username is not present get stored properties
-      ...(!baseInput.session || !baseInput.username ? this.session : null)
+      ...(!baseInput.session || !baseInput.username
+        ? await this._getFromStorage()
+        : null)
     };
     return this.client
```

Comment:
If you are using async/await in the code now, can you also remove the `.then` and `.catch` use?

Reply:
we could, we just have to add a try-catch around it too. think that's a worthwhile pattern to follow for any specific reason?

Reply:
I think that `try/catch` is easier to understand than `.then`/`.catch`. I think it would be a simpler method after the switch. If you've tried it and you think the method is more complicated with `try/catch` then I'll defer to you